osgp/shared/shared/src/test/java/org/opensmartgridplatform/shared/utils/CsvWriterTest.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2019 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.shared.utils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.opensmartgridplatform.shared.utils.csv.CsvWriter;
+
+public class CsvWriterTest {
+
+    private static final String COMMA = ",";
+
+    @Rule
+    public final TemporaryFolder folder = new TemporaryFolder();
+
+    private final CsvWriter csvWriter = new CsvWriter();
+
+    private final String fileName = "file.csv";
+    private String[] header;
+    private List<String[]> lines;
+
+    private String filePath;
+
+    @Before
+    public void setup() throws IOException {
+        this.header = new String[] { "HEADER_1", "HEADER_2" };
```

Comment:
This initialization can also be done as part of the header declaration line (line 38).